src/meta-srv/src/selector/weighted_choose.rs

```diff
@@ -92,6 +92,10 @@ where
     }
 
     fn choose_multiple(&mut self, amount: usize) -> Result<Vec<Item>> {
+        if amount >= self.items.len() {
+            return Ok(self.items.iter().map(|item| item.item.clone()).collect());
```

Comment:
It seems that the items are not actually choosed according to their weights.

Reply:
If the `amount` equals the count of items, then the only possibility is to get all the items.

Reply:
The method `sample_efraimidis_spirakis` said: 
```
/// Randomly sample exactly `amount` distinct indices from `0..length`, and
/// return them in an arbitrary order (there is no guarantee of shuffling or
/// ordering). The weights are to be provided by the input function `weights`,
/// which will be called once for each index.
```

I think it mean:
The `sample_efraimidis_spirakis` function ensures that each index is selected only once through the algorithm when amount <= length. Therefore, when amount == length, the final result will be all unique indices within the range 0..length without duplicates.

Reply:
The method also said:

```
/// This implementation uses `O(length + amount)` space and `O(length)` time
/// if the "nightly" feature is enabled, or `O(length)` space and
/// `O(length + amount * log length)` time otherwise.
```

So I think we can give a fast path here(when the `amount` > items.len).

Reply:
The fast path code has already been removed. Perhaps we shouldn't optimize prematurely before any issues arise. cc @fengys1996